test/augmentation/test_augmentation.py

```diff
@@ -25,6 +25,334 @@
+        raise NotImplementedError("Implement a stupid routine.")
+
+    def test_batch(self):
+        raise NotImplementedError("Implement a stupid routine.")
+
+    def test_jit(self):
+        raise NotImplementedError("Implement a stupid routine.")
+
+    def test_gradcheck(self):
+        raise NotImplementedError("Implement a stupid routine.")
+
+    def test_sequential(self, device, dtype):
+        raise NotImplementedError("Implement a stupid routine.")
+
+    def test_random_batch_transform(self, device, dtype):
+        raise NotImplementedError("Implement a stupid routine.")
+
+# TODO Implement
+# test_random_p_half_return_transform (manual force [True, False] for batchsize 2 inputs)
+# test_batch
+# test_batch_return_transform
+# test_sequential
+# test_sequential_return_transform
+# test_coordinate check
+# test_jit
+# test_gradcheck
+
+    def _create_augmentation_from_params(self, **params):
+        return self._augmentation_cls(**params)
+
+    ############################################################################################################
+    # Test case implementations
+    ############################################################################################################
+
+    def _test_smoke_implementation(self, device, dtype, **params):
+        assert issubclass(self._augmentation_cls,
+                          AugmentationBase2D), f"{self._augmentation_cls} is not a subclass of AugmentationBase2D"
+
+        # Can be instatiated
+        augmentation = self._create_augmentation_from_params(**params, return_transform=False)
+        assert issubclass(
+            type(augmentation), AugmentationBase2D), f"{type(augmentation)} is not a subclass of AugmentationBase2D"
+
+        # generate_parameters can be called and returns the correct amount of parameters
+        batch_shape = (4, 3, 5, 6)
+        generated_params = augmentation.generate_parameters(batch_shape)
+        assert isinstance(generated_params, dict)
+        generated_params["batch_prob"] = torch.tensor([True, True, True, True])
+        for key, value in generated_params.items():
+            assert value.shape[0] == batch_shape[0], f"Value for {key} must have {batch_shape[0]} at position 0 insted of {value.shape[0]}"
+
+        # compute_transformation can be called and returns the correct shaped transformation matrix
+        expected_transformation_shape = torch.Size((batch_shape[0], 3, 3))
+        test_input = torch.ones(batch_shape, device=device, dtype=dtype)
+        transformation = augmentation.compute_transformation(test_input, generated_params)
+        assert transformation.shape == expected_transformation_shape
+
+        # apply_transform can be called and returns the correct batch sized output
+        output = augmentation.apply_transform(test_input, generated_params)
+        assert output.shape[0] == batch_shape[0]
+
+    def _test_smoke_call_implementation(self, device, dtype, **params):
```

Comment:
would be too heavy to call 4 times the same augmentation here ?  same for others

Reply:
test time is already large

Reply:
Do you mean, that I should merge the 3 smoke tests?